Source/Order.swift

```diff
@@ -36,34 +36,112 @@ final public class Order: NSObject, Codable {
 
     /**
      The total order value in pennies (e.g. 3999 for $39.99)
-     or the smallest decimal unit of the currency.
+     or the smallest decimal unit of the currency. (default is 0)
      */
-    private(set) var amount: Int64
+    @available(*, deprecated)
+    let amount: Int64
 
     /**
      The ISO 4217 currency code (default is USD).
      */
-    private(set) var currencyCode: String
+    let currencyCode: String
+    
+    /**
+     The purchase date for the order.
+     */
+    let purchaseDate: Date?
+    
+    /**
+     The button source token
+     */
+    let sourceToken: String?
+    
+    /**
+     The customer-facing order id
+     */
+    let customerOrderId: String?
+    
+    /**
+     A list of the line item details that comprise the order
+     */
+    let lineItems: [LineItems]?
 
+    /**
+     The customer related to the order
+     */
+    let customer: Customer?
+    
+    /**
+     Represents a line item in the order
+     */
+    public class LineItems: NSObject, Codable {
+        
+    }
+    
+    /**
+     Represents a customer in the order
+     */
+    public class Customer: NSObject, Codable {
+        
+    }
+    
     /**
      Initializes an order object with the passed parameters.
 
      - Parameters:
         - id: The order identifier (required).
         - amount: The total order value in pennies or the
                   smallest decimal unit of the currency (e.g. 3999 for $39.99).
+                  Deprecated
         - currencyCode: The ISO 4217 currency code (default is USD).
      */
+    @available(*, deprecated, message: "No longer supported")
     @objc public init(id: String, amount: Int64 = 0, currencyCode: String = "USD") {
         self.id = id
         self.amount = amount
         self.currencyCode = currencyCode
+        self.purchaseDate = nil
+        self.sourceToken = nil
+        self.customerOrderId = nil
+        self.lineItems = nil
+        self.customer = nil
+    }
+    
+    /**
+     Initializes an order object with the passed parameters.
+     
+     - Parameters:
+     - id: The order identifier (required).
+     - amount: The total order value in pennies or the
+               smallest decimal unit of the currency (e.g. 3999 for $39.99).
+               Deprecated
+     - currencyCode: The ISO 4217 currency code (default is USD).
+     - purchaseDate: The date of the purchase for the order.
+     - sourceToken: The customer-facing order identifer.
+     - customerOrderId: The customer-face order Id.
+     - lineItems: A list of the line item details that comprise the order.
+     - customer: Customer field
+     */
+    @objc public init(id: String, amount: Int64 = 0, currencyCode: String = "USD", purchaseDate: Date? = nil, sourceToken: String? = nil, customerOrderId: String? = nil, lineItems: [LineItems]? = nil, customer: Customer? = nil) {
```

Comment:
In this new init  `purchaseDate` and `lineItems` should be required and not nullable to enforce the required fields on the api